### PR TITLE
Add multiple sequelize instance support

### PR DESCRIPTION
**Pull request by aszx87410**
Add optional parameter `sequelize` to `epilogue.resource` . It allow users to use different sequelize instance.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -46,6 +46,16 @@ var epilogue = {
     if (!options.model)
       throw new Error('please specify a valid model');
```
Comment (mbroadst) [inline lib/index.js]: nitpick, I would probably just rewrite this as:
```
  options.sequelize = ensureSequelizeInstance(options.sequelize || this.sequelize);
```

and refactor the check into a function at the top of file:
```
function ensureSequelizeInstance(maybeInstance) {
  var instance = (maybeInstance.Sequelize) ? maybeInstance.Sequelize : maybeInstance;
  if (!_.every(requiredSequelizeAttrs, _.partial(_.has, instance)))
    throw new Error('invalid sequelize instance');

  return instance;
}
```

**Comment by mbroadst** [comment]
I note you chose the easy thing to test here :smile: Can you please add a test that creates a resource that connects to a different sequelize instance?

**Comment by aszx87410** [comment]
I appreciate your feedback.
After spent a lot of time, I still can't figure out how to add test.
How to know if a resource really connect to the different instance?

I've tried the following way, but it didn't work.
```
it('should allow users to use different sequelize instance', function() {
      
      var db = new test.Sequelize('minor', null, null, {
        dialect: 'sqlite',
        storage: ':memory:',
        logging: (process.env.SEQ_LOG ? console.log : false)
      });

      var Book = db.define('books', {
        id: { type: test.Sequelize.INTEGER, autoIncrement: true, primaryKey: true },
        name: { type: test.Sequelize.STRING },
        author: { type: test.Sequelize.STRING }
      }, {
        underscored: true,
        timestamps: false,
      });

      var resource = rest.resource({
        model: Book,
        sequelize: db
      });

      expect(rest.sequelize).to.not.equal(resource.sequelize);
    });
```
any suggestion?

**Comment by mbroadst** [comment]
@aszx87410 if that test doesn't work, then I don't believe your patch works as intended. That looks like a fine test for verifying the behavior. You might want to change the assertion to a deep equality like so: `expect(rest.sequelize).to.not.eql(resource.sequelize);`

**Comment by aszx87410** [comment]
@mbroadst I think the problem is here: https://github.com/dchester/epilogue/blob/master/lib/index.js#L22
`this.sequelize = (options.sequelize.Sequelize) ?
      options.sequelize.Sequelize : options.sequelize;`

No matter two sequelize instance equals or not, `sequelizeInstance.Sequelize` is always equal.
refer to: http://docs.sequelizejs.com/en/latest/api/sequelize/#sequelize
`sequelizeInstanceA.Sequelize===sequelizeInstanceB.Sequelize` is always true.

And because of this, I founded that I can directly connect two different database by following code:
```
// User = db1.define(...), which is from database1 
var userResource = epilogue.resource({
  model: User,
  endpoints: ['/users', '/users/:id']
});

// U2 = db2.define(...), which is from database2
var u2 = epilogue.resource({
	model: U2,
	endpoints: ['/user2', '/user2/:id']
})
```
It's needless to pass different sequelize instance, just pass the model from different sequelize instance.
If i have said something wrong, please correct my mistakes. Thanks! :smile:

**Comment by mbroadst** [comment]
@aszx87410 sounds good to me, so we don't even need this patch then huh?

**Comment by aszx87410** [comment]
Yep. Thanks for your kind help.

**Pull request closed without merging**